### PR TITLE
[dtensor] add placement._shard_tensor for DTensor.from_local

### PR DESCRIPTION
**Pull request by guoyejun**
test code:
    # create a DistributedTensor that shards on dim 0, from a local torch.Tensor
    rowwise_placement=[Shard(0)]
    local_tensor = torch.randn((8, 8), requires_grad=True)
    rowwise_tensor = DTensor.from_local(local_tensor, device_mesh, rowwise_placement)

the expecation is that rowwise_tensor is sharded with row wise

cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @kwen2501 @awgu @fegin @XilunWu @wanchaol @fduwjj @wz337 @tianyu-l @wconstab @chauhang @d4l3k

### CODE REVIEW

**Comment by guoyejun** [comment]
looks that it is my mistake, closing the PR.
